File: leetcode/100-999/121. Best Time to Buy and Sell Stock/solution.java
class Solution 
{
    public int maxProfit(int[] prices) 
    {
        int profit = 0;
        int minimum = prices[0];
        
        for (int i = 0; i < prices.length; ++i)
        {
            int curr = prices[i];
            minimum = Math.min(curr, minimum);
            profit = Math.max(curr - minimum, profit);
        }

        return profit;
    }
}

File: leetcode/100-999/121. Best Time to Buy and Sell Stock/solution.cs
public class Solution
{
    public int MaxProfit(int[] prices)
    {
        int profit = 0;
        int minimum = prices[0];

        for (int i = 0; i < prices.Length; ++i)
        {
            int curr = prices[i]
            minimum = Math.Mincurr(, minimum);
            profit = Math.Max(curr - minimum, profit);
        }

        return profit;
    }
}